util/internal/test/common/src/main/java/org/hibernate/search/util/impl/test/extension/parameterized/ParameterizedClass.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.util.impl.test.extension.parameterized;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+import org.junit.jupiter.api.TestInstance;
+
+/**
+ * A marker annotation to let JUnit engine know how to treat the test class annotated with it.
+ * <p>
+ * Tests that are expected to run as parameterized classes must have a setup method annotated with {@link ParameterizedSetup}.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public @interface ParameterizedClass {
```

Comment:
Could you somehow add a check for classes annotated with this, so that an error is thrown if there is no `ParameterizedSetup`-annotated method?

Because that would be a pretty dangerous configurations: tests would be filtered and nothing would replace them...

I'm guessing this could be done through another extension added by this annotation.

Reply:
In some cases, an Extension might be too late to detect problems in the test config, I've already had some checks in the `ParameterizedClassPostDiscoveryFilter` so I'll expend those and make them more "visible" 😃

Reply:
BTW... if we do the checks in the filter and throw an exception, the entire test suite will fail at the very beginning, so we wouldn't need to wait until the end to learn that we have a misconfigured test.